stdlib/public/Cxx/CxxConvertibleToCollection.swift

```diff
@@ -63,10 +63,10 @@ extension Set {
   ///   container when each element is copied in O(1). Note that this might not
   ///   be true for certain C++ types, e.g. those with a custom copy
   ///   constructor that performs additional logic.
-  public init<C: CxxConvertibleToCollection>(_ c: C)
+  public init<C: CxxConvertibleToCollection>(contentsOf c: C)
     where C.RawIterator.Pointee == Element {
 
     self.init()
-    forEachElement(of: c) { self.insert($0) }
+    c.forEach { self.insert($0) }
```

Comment:
Two recommendations:

1. Instead of just extending `Set`, add this as a generic extension on `SetAlgebra` protocol.
2. Same idea about `reserveCapacity` as with the `RangeReplaceableCollection` extension above. (With the same caveats, too.)

```swift
extension SetAlgebra {
  public init(contentsOf c: some CxxConvertibleToCollection<Element>) {
    self.init()
    self.reserveCapacity(c.count)
    c.forEach { self.insert($0) }
  }
}
```

Reply:
> Instead of just extending Set, add this as a generic extension on SetAlgebra protocol.

Done!